CMakeLists.txt

```
@@ -48,19 +48,134 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
```

I think we need here to check also for MIXXX_VCPKG_ROOT if that exists, we are likely in an already configured build directory and can skip the whole block. 
```suggestion
if(WIN32 AND IS_DIRECTORY MIXXX_VCPKG_ROOT)
```

---

This would no longer detect if someone changed the buildenv to use by specifing -DBUILDENV_NAME The typical use case when you download a buildenv ZIP file from a VCPKG PR.

---

Ah I see. However it looks like id fails now when BUILDENV_NAME  and BUILDENV_URL is not even though if MIXXX_VCPKG_ROOT is valid. I consider BUILDENV_NAME and BUILDENV_URL as internal, used for communication form the batch file. Does that make sense?

---

You mean the case that somebody locally builds VCPKG and just set MIXXX_VCPKG_ROOT ? In all other cases BUILDENV_NAME  or BUILDENV_URL are the leading vars, where MIXXX_VCPKG_ROOT could be derived from. And if windows_buildenv.bat is used, all vars are in sync anyway.

---

This is the new situation now after this PR. If one uses a branch from before only MIXXX_VCPKG_ROOT is used by cmake. "VCPKG_ROOT" is the official upstream way to point to the build environment. We have introduced MIXXX_VCPKG_ROOT for the same purpose because "VCPKG_ROOT" is already set to a different directory on the GitHub workflow runners. 

That's why I think it is more intuitive to keep "MIXX_VCPKG_ROOT" for the case you want to switch between two environments without issuing a download. I consider BUILDENV_NAME and BUILDENV_URL as programmatic script helper the user should not set manually.

---

No it's the other way around, MIXXX_VCPKG_ROOT is just a a concatenation from BUILDENV_BASEPATH and BUILDENV_NAME inside windows_buildenv.bat. I will move this concatenation from windows_buildenv.bat to CMakeLists.txt than we have a straight forward data flow without duplicated env variables.
The basepath should be overwriteable for it's own, to allow a central storage location used by all worktrees.